src/core/pointcloud/qgspointcloudlayerprofilegenerator.h

```diff
@@ -0,0 +1,163 @@
+/***************************************************************************
+                         qgspointcloudlayerprofilegenerator.h
+                         ---------------
+    begin                : April 2022
+    copyright            : (C) 2022 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSPOINTCLOUDLAYERPROFILEGENERATOR_H
+#define QGSPOINTCLOUDLAYERPROFILEGENERATOR_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsabstractprofilegenerator.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgscoordinatetransformcontext.h"
+#include "qgscoordinatetransform.h"
+#include "qgslinesymbol.h"
+#include "qgsvector3d.h"
+
+#include <memory>
+
+class QgsProfileRequest;
+class QgsCurve;
+class QgsPointCloudLayer;
+class QgsAbstractTerrainProvider;
+class QgsProfileSnapContext;
+class QgsPointCloudRenderer;
+class IndexedPointCloudNode;
+class QgsPointCloudIndex;
+class QgsPointCloudRequest;
+class QgsPointCloudBlock;
+class QgsGeos;
+
+#define SIP_NO_FILE
+
+
+/**
+ * \brief Implementation of QgsAbstractProfileResults for point cloud layers.
+ *
+ * \note Not available in Python bindings
+ * \ingroup core
+ * \since QGIS 3.26
+ */
+class CORE_EXPORT QgsPointCloudLayerProfileResults : public QgsAbstractProfileResults
+{
+
+  public:
+
+    struct PointResult
+    {
+      double x;
+      double y;
+      double z;
+      double distance;
+      double curveDistance;
```

Comment:
from a quick look it's unclear which distance means distance along the curve, and which one is the distance away from the curve? maybe a comment or renaming would help